src/machine/machine_atmega.go

```diff
@@ -143,38 +165,37 @@ func (uart *UART) Configure(config UARTConfig) {
 		config.BaudRate = 9600
 	}
 
-	// Register the UART interrupt.
-	interrupt.New(irq_USART0_RX, func(intr interrupt.Interrupt) {
-		// Read register to clear it.
-		data := avr.UDR0.Get()
-
-		// Ensure no error.
-		if !avr.UCSR0A.HasBits(avr.UCSR0A_FE0 | avr.UCSR0A_DOR0 | avr.UCSR0A_UPE0) {
-			// Put data from UDR register into buffer.
-			UART0.Receive(byte(data))
-		}
-	})
-
 	// Set baud rate based on prescale formula from
 	// https://www.microchip.com/webdoc/AVRLibcReferenceManual/FAQ_1faq_wrong_baud_rate.html
 	// ((F_CPU + UART_BAUD_RATE * 8L) / (UART_BAUD_RATE * 16L) - 1)
 	ps := ((CPUFrequency()+config.BaudRate*8)/(config.BaudRate*16) - 1)
-	avr.UBRR0H.Set(uint8(ps >> 8))
-	avr.UBRR0L.Set(uint8(ps & 0xff))
+	uart.baudRegH.Set(uint8(ps >> 8))
+	uart.baudRegL.Set(uint8(ps & 0xff))
 
 	// enable RX, TX and RX interrupt
-	avr.UCSR0B.Set(avr.UCSR0B_RXEN0 | avr.UCSR0B_TXEN0 | avr.UCSR0B_RXCIE0)
+	uart.statusRegB.Set(avr.UCSR0B_RXEN0 | avr.UCSR0B_TXEN0 | avr.UCSR0B_RXCIE0)
 
 	// 8-bits data
-	avr.UCSR0C.Set(avr.UCSR0C_UCSZ01 | avr.UCSR0C_UCSZ00)
+	uart.statusRegC.Set(avr.UCSR0C_UCSZ01 | avr.UCSR0C_UCSZ00)
+}
+
+func (uart *UART) handleInterrupt(intr interrupt.Interrupt) {
+	// Read register to clear it.
+	data := uart.dataReg.Get()
+
+	// Ensure no error.
+	if !uart.statusRegA.HasBits(avr.UCSR0A_FE0 | avr.UCSR0A_DOR0 | avr.UCSR0A_UPE0) {
```

Comment:
Can look confusing we use constants *associated* with UART0 here, but they are actually the same for all UARTs. Keeps the code size smaller and cleaner.
Same comment for several lines above.

Reply:
Understood, that is not uncommon with the generated wrappers from the SVD files.